src/key.rs

```diff
@@ -26,7 +27,22 @@ use Verification;
 use constants;
 use ffi::{self, CPtr};
 
-/// Secret 256-bit key used as `x` in an ECDSA signature
+/// Secret 256-bit key used as `x` in an ECDSA signature.
+///
+/// # Examples
+///
+/// Basic usage:
+///
+/// ```
+/// # #[cfg(feature="rand")] {
+/// use secp256k1::{Secp256k1, SecretKey};
+/// use secp256k1::rand::rngs::OsRng;
+///
+/// let secp = Secp256k1::new();
+/// let mut rng = OsRng::new().expect("OsRng");
```

Comment:
Hmm why not `thread_rng()`?

Reply:
Because we use `thread_rng` everywhere else, I thought it might be nice to use OsRng here, although its already used in `lib.rs` example so doesn't really add that much.